petastorm/pytorch.py

```diff
@@ -150,6 +144,22 @@ def __iter__(self):
 
         keys = None
 
+        if self._in_iter:
+            raise RuntimeError('Only after previous iteration finished we can start another iteration.')
```

Comment:
minor: This is user-facing error message but it doesn't provide any instructions. How about:

~~~
You must finish a full pass of Petastorm DataLoader before making another pass from the beginning.
If you do need to terminate early and restart from beginning, please re-create the reader and the data loader.
~~~